cluster/engine.go

```diff
@@ -182,45 +203,114 @@ func (e *Engine) isConnected() bool {
 
 // IsHealthy returns true if the engine is healthy
 func (e *Engine) IsHealthy() bool {
-	return e.healthy
+	e.RLock()
+	e.RUnlock()
+	return e.state == stateHealthy
+}
+
+// setState sets engine state
+func (e *Engine) setState(state engineState) {
+	e.Lock()
+	defer e.Unlock()
+	e.state = state
 }
 
-// setHealthy sets engine healthy state
-func (e *Engine) setHealthy(state bool) {
+// TimeToValidate returns true if a pending node is up for validation
+func (e *Engine) TimeToValidate() bool {
 	e.Lock()
-	e.healthy = state
-	// if engine is healthy, clear failureCount
-	if state {
-		e.failureCount = 0
+	defer e.Unlock()
+	if e.state != statePending {
+		return false
 	}
-	e.Unlock()
+	sinceLastUpdate := time.Since(e.updatedAt)
+	// Increase check interval for a pending engine according to failureCount and cap it at 4 hours
+	if sinceLastUpdate > 4*time.Hour || sinceLastUpdate > time.Duration(e.failureCount)*30*time.Second {
```

Comment:
small nit but I feel like `4*time.Hour` and `30*time.Second` should be constants here.

Suggestions:

`validationLimit -> 4*time.hour`
`retryFactor -> 30*time.Second`